### DISCUSSION TITLE
Application can't be viewed in design view

### DISCUSSION THREAD

**Discussion by sovzzz** [AvaloniaUI/Avalonia#13888] [Q&A]
when I tried to preview in VSCode, it says "Application can't be viewed in design view."
tried in VS, it warns "Invalid Markups". log output by Avalonia diagnostic is below.
```14:44:22.944 [Information]  Avalonia Package initialized
14:44:33.462 [Information] 0 Starting previewer process for '"D:\Launcher\project\Aki.Launcher\bin\Debug\net6.0\win10-x64\Aki.Launcher.dll"'
14:44:33.483 [Information] 2936 Started previewer process for '"D:\Launcher\project\Aki.Launcher\bin\Debug\net6.0\win10-x64\Aki.Launcher.dll"'. Waiting for connection to be initialized.
14:44:33.629 [Information] 2936 Connection initialized
14:44:34.735 [Error] 2936 UpdateXamlResult error
System.Xaml.XamlException: The type initializer for 'Aki.Launcher.Helpers.LocalizationProvider' threw an exception.
```

the code can be found here https://dev.sp-tarkov.com/SPT-AKI/Launcher/src/branch/master/project/Aki.Launcher.Base/Helpers/LocalizationProvider.cs

**Comment by maxkatz6** [comment]
You can either attach a debugger from your LocalizationProvider method or add try/catch with logging there so you can find the issue in the code.

**Comment by thevortexcloud** [comment]
The previewer works by running your actual app. As such it's fairly easy for it to break if you are using a design instance of your view model, and your code relies on other things such as injected values in your view model or accessing data based on specific configurations. But since this is a static class I would guess one of your static methods/variables is failing. Possibly because `CurrentDirectory` is different. I am not sure what the previewer sets that to. In any case you can use `Design.IsDesignMode` to detect when you are using the previewer and do somehthing else (EG load dummy data).
